pkg/crc/preflight/preflight_checks_tray_windows.go

```diff
@@ -0,0 +1,206 @@
+	if err != nil {
+		return fmt.Errorf("Failed to install CodeReady Containers daemon service: %v", err)
+	}
+	return nil
+}
+
+func removeDaemonService() error {
+	err := service.Stop(constants.DaemonServiceName)
+	if err != nil {
+		return fmt.Errorf("Failed to stop the daemon service: %v", err)
+	}
+	return service.Delete(constants.DaemonServiceName)
+}
+
+func checkIfDaemonServiceRunning() error {
+	if service.IsRunning(constants.DaemonServiceName) {
+		return nil
+	}
+	return fmt.Errorf("CodeReady Containers dameon service is not running")
+}
+
+func fixDaemonServiceRunning() error {
+	return service.Start(constants.DaemonServiceName)
+}
+
+func checkTrayBinaryExists() error {
+	if os.FileExists(constants.TrayBinaryPath) {
+		return nil
+	}
+	return fmt.Errorf("Tray binary does not exists")
+}
+
+func fixTrayBinaryExists() error {
+	tmpArchivePath, err := ioutil.TempDir("", "crc")
+	if err != nil {
+		logging.Error("Failed creating temporary directory for extracting tray")
+		return err
+	}
+	defer func() {
+		_ = goos.RemoveAll(tmpArchivePath)
+	}()
+
+	logging.Debug("Trying to extract tray from crc binary")
+	err = embed.Extract(filepath.Base(constants.GetCRCWindowsTrayDownloadURL()), tmpArchivePath)
+	if err != nil {
+		logging.Debug("Could not extract tray from crc binary", err)
+		logging.Debug("Downloading crc tray")
+		_, err = dl.Download(constants.GetCRCWindowsTrayDownloadURL(), tmpArchivePath, 0600)
+		if err != nil {
+			return err
+		}
+	}
+	archivePath := filepath.Join(tmpArchivePath, filepath.Base(constants.GetCRCWindowsTrayDownloadURL()))
+	outputPath := constants.CrcBinDir
+	err = goos.MkdirAll(filepath.Join(outputPath, "tray-windows"), 0750)
+	if err != nil && !goos.IsExist(err) {
+		return fmt.Errorf("Cannot create the target directory: %v", err)
+	}
+	err = extract.Uncompress(archivePath, filepath.Join(outputPath, "tray-windows"))
+	if err != nil {
+		return fmt.Errorf("Cannot uncompress '%s': %v", archivePath, err)
+	}
+
+	// If a tray is already running kill it
+	if err := checkTrayRunning(); err == nil {
+		cmd := `Stop-Process -Name "tray-windows"`
+		if _, _, err := powershell.Execute(cmd); err != nil {
+			logging.Debugf("Failed to kill running tray: %v", err)
+		}
+	}
+	return nil
+}
+
+func checkTrayBinaryVersion() error {
+	versionCmd := `(Get-Item %s).VersionInfo.ProductVersion`
+	stdOut, stdErr, err := powershell.Execute(fmt.Sprintf(versionCmd, constants.TrayBinaryPath))
+	if err != nil {
+		return fmt.Errorf("Failed to get the version of tray: %v: %s", err, stdErr)
+	}
+	currentTrayVersion := strings.TrimSpace(stdOut)
+	if currentTrayVersion != version.GetCRCWindowsTrayVersion() {
+		return fmt.Errorf("Current tray version doesn't match with expected version")
+	}
+	return nil
+}
+
+func fixTrayBinaryVersion() error {
+	return fixTrayBinaryExists()
+}
+
+func checkTrayBinaryAddedToStartupFolder() error {
+	if os.FileExists(filepath.Join(startUpFolder, constants.TrayShortCutName)) {
```

Comment:
Shortcut is a singleword. No camelcase needed

Reply:
Changed, thanks!